xinference/core/model.py

```diff
@@ -109,24 +136,38 @@ async def __pre_destroy__(self):
             gc.collect()
             torch.cuda.empty_cache()
 
-    def __init__(self, model: "LLM"):
+    def __init__(self, model: "LLM", request_limits: Optional[int] = None):
         super().__init__()
         from ..model.llm.pytorch.core import PytorchModel
         from ..model.llm.pytorch.spec_model import SpeculativeModel
         from ..model.llm.vllm.core import VLLMModel
 
         self._model = model
+        self._request_limits = request_limits
 
         self._generators: Dict[str, Union[Iterator, AsyncGenerator]] = {}
         self._lock = (
             None
             if isinstance(self._model, (PytorchModel, SpeculativeModel, VLLMModel))
             else asyncio.locks.Lock()
         )
+        self._serve_count = 0
+        self._limit_lock = asyncio.locks.Lock()
 
     def load(self):
         self._model.load()
 
+    def model_uid(self):
+        return (
+            self._model.model_uid
+            if hasattr(self._model, "model_uid")
+            else (
+                self._model._model_uid
+                if hasattr(self._model, "_model_uid")
```

Comment:
Why both of `model_uid` and `_model_uid` are checked here?

Reply:
> Why both of `model_uid` and `_model_uid` are checked here?

Because `self._model` can be `EmbeddingModel` or `LLMModel`, `EmbeddingModel` has the attribute `_model_uid`,  and `LLMModel` has the attribute `model_uid`.